### PR TITLE
chore: rename packages to hiero

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:

- [x] Rename packages to org.hiero
- [x] Switch licenses to SPDX-License
- [x] spotlessApply

**Related issue(s)**: https://github.com/hashgraph/hedera-sdk-java/issues/2017

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by 0xivanov** [comment]
Hey @jjohannes , do you think it's possible to rename the packages to without breaking the `example` module?

**Comment by jjohannes** [comment]
Renaming a package is technically changing the API. To be consistent, we need to rename everything at once in my opinion:

- The packages in `sdk`
  - All usages of classes from the packages in tests and examples
- The Java Module Names [here](https://github.com/hashgraph/hedera-sdk-java/blob/0a72de56828cc2913b155d97225ccad1534086eb/sdk/src/main/java/module-info.java#L21) and [here](https://github.com/hashgraph/hedera-sdk-java/blob/0a72de56828cc2913b155d97225ccad1534086eb/sdk-full/src/main/java/module-info.java#L21)
- The **group** for the Maven Coordinates [here](https://github.com/hashgraph/hedera-sdk-java/blob/0a72de56828cc2913b155d97225ccad1534086eb/settings.gradle.kts#L7-L9) (and [here](https://github.com/hashgraph/hedera-sdk-java/blob/0a72de56828cc2913b155d97225ccad1534086eb/.github/workflows/release-artifacts.yml#L132) in publishing CI config).

Users that switch to the new Hiero version of the SDK will have to adjust there imports to the new package(s). But the classes themselves will work as before. Only the imports need to be adjusted.

**Comment by 0xivanov** [comment]
@jjohannes I don't know why the examples are not building, i haven't touched them. Do you have any ideas?

File: sdk-full/src/main/java/module-info.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *
  */
 
-module com.hedera.hashgraph.sdk.full {
+module com.hiero.sdk.full {
```
Comment (jjohannes) [inline sdk-full/src/main/java/module-info.java:21]: I think the new namespace should be `org.hiero`.

**Comment by jjohannes** [comment]
> I don't know why the examples are not building

With the rename, the examples now automatically use the last published version of the `sdk` (2.46.0).

That version is without the recent Guava/CheckerQual version update. But the build setup used for the example expects a newer version due to our strict Java Module setup. The newer version has a `module-info` and this is now expected.

You can update the Guava version in the example directly – [dependency constraints block](https://github.com/hashgraph/hedera-sdk-java/blob/0a72de56828cc2913b155d97225ccad1534086eb/examples/build.gradle.kts#L20-L25) – to make it work with the previous version:
(Not this is only an issue in our strict checking setup, for users of the SDK without these strict checks, the example works)

```
dependencies.constraints {
    implementation("com.google.guava:guava:33.3.1-android")
    ...
}
```

**Review by mishomihov00 — APPROVED** [review]
Review applies to:
`.github/workflows/build.yml`
`.github/workflows/release-artifacts.yml`

File: tck/build.gradle.kts [hunk 2]
```diff
@@ -63,6 +62,7 @@ dependencyAnalysis {
         onAny {
             severity("fail")
             exclude("com.google.protobuf:protobuf-javalite")
+            onUnusedDependencies { exclude(":sdk") }
```
Comment (jjohannes) [inline tck/build.gradle.kts:65]: Why is this needed?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -101,7 +101,7 @@ jobs:
           OPERATOR_ID: '0.0.1022'
           HEDERA_NETWORK: 'localhost'
         run: |
-          ./gradlew -POPERATOR_ID=$OPERATOR_ID -POPERATOR_KEY=$OPERATOR_KEY -PHEDERA_NETWORK=$HEDERA_NETWORK :aggregation:testCodeCoverageReport --scan
+          ./gradlew -POPERATOR_ID=$OPERATOR_ID -POPERATOR_KEY=$OPERATOR_KEY -PHEDERA_NETWORK=$HEDERA_NETWORK :aggregation:testCodeCoverageReport --scan --rerun-tasks
```
Comment (jjohannes) [inline .github/workflows/build.yml:104]: The `--rerun-tasks` should not be added here. It will lead to recompiling the code and not reuse it from the step before. It will also not use anything from the build cache making the remote cache more or less useless. What is the reason for adding it?

File: tck/build.gradle.kts (hunk 2 above)
Comment (0xivanov) [reply tck/build.gradle.kts:65]: some check was failing before, let's see if it's fixed now

File: .github/workflows/build.yml (hunk 3 above)
Comment (0xivanov) [reply .github/workflows/build.yml:104]: Alright

Comment (0xivanov) [reply .github/workflows/build.yml]: I just wanted to make sure the tests are actually running

File: tck/build.gradle.kts (hunk 2 above)
Comment (0xivanov) [reply tck/build.gradle.kts]: the healthcheck fails, not sure why.

Comment (jjohannes) [reply tck/build.gradle.kts]: It should not fail, but it does... Somehow the rename must confuse it. If you want you can keep it in and I will investigate later.

Comment (0xivanov) [reply tck/build.gradle.kts]: alright

**Review by jjohannes — DISMISSED** [review]

**Comment by jjohannes** [comment]
The package for the protobuf-generated classes should probably also be changed:

See: https://github.com/hashgraph/hedera-sdk-java/blob/main/sdk/src/main/proto/account.proto#L28-L29
And all other `proto` files

The content of `sdk/build/classes/java/main` after building, shows if you still have any `com.hedera` classes.

**Review by jjohannes — APPROVED** [review]

**Review by rbarkerSL — APPROVED** [review]
Approval applies to:
- .github/workflows/release-artifacts.yml
- example-android/app/build.gradle.kts
- example-android/build.gradle.kts
- example-android/settings.gradle.kts
- examples/build.gradle.kts
- settings.gradle.kts
- tck/build.gradle.kts

**Review by agadzhalov — APPROVED** [review]
LGTM, nice work, tyvm @0xivanov @jjohannes
